From f444a316f5178b8e20fe31e7b2d979e651da077e Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyich@gmail.com>
Date: Sat, 23 Dec 2023 19:12:42 +0000
Subject: [PATCH 1/2] configure.ac: fix `autoconf-2.72` compatibility

`autoconf-2.72` slightly changed `AS_IF`/`AC_CHECL_LIB` definitions and
exposed the bug of missng quoting around the arguments:

    editline> ./configure: line 13944: syntax error near unexpected token `;;'
    editline> ./configure: line 13944: ` ;;'

The change adds quoting as suggested by https://savannah.gnu.org/support/index.php?110990
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 880f6f1..3fe0410 100644
--- a/configure.ac
+++ b/configure.ac
@@ -89,7 +89,7 @@ AS_IF([test "x$enable_terminal_bell" = "xyes"],
 AM_CONDITIONAL([ENABLE_EXAMPLES], [test "$enable_examples" = yes])
 
 # Check for a termcap compatible library if enabled
-AS_IF([test "x$enable_termcap" = "xyes"],
+AS_IF([test "x$enable_termcap" = "xyes"], [
    AC_DEFINE(CONFIG_USE_TERMCAP, 1, [Define to use the termcap library for terminal size.])
    AC_CHECK_LIB(terminfo, tgetent, , [
       AC_CHECK_LIB(termcap, tgetent, , [
@@ -100,7 +100,7 @@ AS_IF([test "x$enable_termcap" = "xyes"],
                ])
             ])
          ])
-      ]))
+      ])])
 
 # Generate all files
 AC_OUTPUT

From 2b788be1c8cd7d100da867255ad77d3b43260901 Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyich@gmail.com>
Date: Sat, 23 Dec 2023 19:17:19 +0000
Subject: [PATCH 2/2] configure.ac: add second parameter quoting around the
 rest of AS_IF for consistency

---
 configure.ac | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3fe0410..2a01fad 100644
--- a/configure.ac
+++ b/configure.ac
@@ -68,23 +68,23 @@ AC_ARG_ENABLE([examples],
 #
 # Check what features have been enabled
 #
-AS_IF([test "x$enable_unique_history" != "xno"],
-   AC_DEFINE(CONFIG_UNIQUE_HISTORY, 1, [Define to skip duplicate lines in the scrollback history.]))
+AS_IF([test "x$enable_unique_history" != "xno"], [
+   AC_DEFINE(CONFIG_UNIQUE_HISTORY, 1, [Define to skip duplicate lines in the scrollback history.])])
 
-AS_IF([test "x$enable_terminal_bell" != "xno"],
-   AC_DEFINE(CONFIG_ANSI_ARROWS, 1, [Define to include ANSI arrow keys support.]))
+AS_IF([test "x$enable_terminal_bell" != "xno"], [
+   AC_DEFINE(CONFIG_ANSI_ARROWS, 1, [Define to include ANSI arrow keys support.])])
 
-AS_IF([test "x$enable_eof" != "xno"],
-   AC_DEFINE(CONFIG_EOF, 1, [Define to enable EOF (Ctrl-D) key.]))
+AS_IF([test "x$enable_eof" != "xno"], [
+   AC_DEFINE(CONFIG_EOF, 1, [Define to enable EOF (Ctrl-D) key.])])
 
-AS_IF([test "x$enable_sigint" != "xno"],
-   AC_DEFINE(CONFIG_SIGINT, 1, [Define to enable SIGINT (Ctrl-C) key.]))
+AS_IF([test "x$enable_sigint" != "xno"], [
+   AC_DEFINE(CONFIG_SIGINT, 1, [Define to enable SIGINT (Ctrl-C) key.])])
 
-AS_IF([test "x$enable_sigstop" = "xyes"],
-   AC_DEFINE(CONFIG_SIGSTOP, 1, [Define to enable SIGSTOP (Ctrl-Z) key.]))
+AS_IF([test "x$enable_sigstop" = "xyes"], [
+   AC_DEFINE(CONFIG_SIGSTOP, 1, [Define to enable SIGSTOP (Ctrl-Z) key.])])
 
-AS_IF([test "x$enable_terminal_bell" = "xyes"],
-   AC_DEFINE(CONFIG_TERMINAL_BELL, 1, [Define to enable terminal bell on completion.]))
+AS_IF([test "x$enable_terminal_bell" = "xyes"], [
+   AC_DEFINE(CONFIG_TERMINAL_BELL, 1, [Define to enable terminal bell on completion.])])
 
 AM_CONDITIONAL([ENABLE_EXAMPLES], [test "$enable_examples" = yes])
 
